sirepo/package_data/static/js/raydata.js

```diff
@@ -107,15 +107,18 @@ SIREPO.app.directive('catalogPicker', function() {
             field: '=',
         },
         template: `
-            <select class="form-control" data-ng-model="model[field]" data-ng-options="name as name for name in catalogNames"></select>
+            <div data-ng-hide="!awaitingCatalogNames">Loading...</div>
```

Comment:
Make the variable something like "haveCatalogNames". Then you can get rid of the `!`. `!` can be confusing (I fall in the group of people it confuses) and should be avoided when possible. It is especially confusing in this case where true means we do want to hide something (I ascribe hide to a "negative"). Hard for me to keep it all straight... 

I can't find a good quote but Steve McConnell in his frequently praised book Code Complete was the first person I saw mention this. Here is a blog that sums it up https://schneide.blog/2014/08/03/dont-ever-not-avoid-negative-logic/

Reply:
I found this comment a little confusing. The line after this used `awaitingCatalogNames` without the `!`, so renaming the variable wouldn't get rid of the need for `!`. But I changed one of the `data-ng-hide` to `data-ng-show` to get rid of the `!`. Is that what you had in mind?

9834229108193197e2be278d92997ffbf798781f

Reply:
Agh I'm sorry I totally misread the code! I didn't see the use below. I think using ng-hide and ng-show is actually more confusing. Better to use the same one and have the variable negated in one case. I don't have a preference on haveCatalogNames vs awaitingCatalogNames. Sorry to make you do this unnecessary work.

Reply:
[da3290d](https://github.com/radiasoft/sirepo/pull/4977/commits/da3290dd1c15209bcdbe06064ce9ba0668b5b172)